src/components/articles/cbd-article.vue

```diff
@@ -0,0 +1,132 @@
+
+<template>
+   <div style="border:none;margin-top:10px">
+        <div v-if="!loading">
+
+            <div v-if="!hideCoverImage && article && article.coverImage && article.coverImage.url">
+                <cbd-article-cover-image cover-image="article.coverImage"></cbd-article-cover-image>
+            </div>
+        
+            <div v-if="hasEditRights" class="pull-right">    
+                <cbd-add-new-article :tags="tags" :admin-tags="adminTags" :custom-tags="customTags" :id="(article||{})._id" :target="target"
+                    class="btn btn-default"></cbd-add-new-article>
+            </div>
+
+            <div v-if="article" v-html="$options.filters.lstring(article.content, $locale)" class="ck-content"></div>
+
+            <slot name="empty-article">
+                <div v-if="!article" class="ck-content">No information is available for this section at the moment.</div>
+            </slot>
+        </div>
+        <slot name="loading-article">
+            <div v-if="loading">Loading content... <i class="fa fa-spinner fa-spin"></i></div>
+        </slot>
+    </div>
+
+</template>
+
+<script>
+
+import 'css!cdn!npm/@scbd/ckeditor5-build-inline-full@35.0.0/build/content-style.css';
+import axios from 'axios';
+import ArticlesApi from '../../services/api/articles';
+import cbdAddNewArticle from './cbd-add-new-article.vue';
+import {lstring } from '../../services/filters/lstring'
+
+export default {
+    name: 'cbdArticle',
+    components : { cbdAddNewArticle },
+    props: {
+        hideCoverImage  : { type: Boolean, required: false, default:false        },
+        showEdit        : { type: Boolean, required: false, default:undefined    },
+        article         : { type: Object,  required: false, default:undefined    },
+        query           : { type: Object,  required: false                       },
+        tags 		    : { type: Array  , required: false, default:[]           }, // [] of tag id's
+        customTags 	    : { type: Array  , required: false, default:[]           }, // [] of customTag id's
+        adminTags 	    : { type: Array  , required: false, default:[]           }, // [] of adminTag text
+        target          : { type: String , required: false, default: '_self'     },
+        editRoles       : { type: Array  , required: false, default:[]           }, // [] of scope roles
+    },
+    data() {
+        return {
+            returnUrl       : window.location.href,
+            hasEditRights   : false,
+            loading         : false
+        }
+    },
+    created() {
+       this.ArticlesApi = new ArticlesApi({ token: this.$auth.strategy.token.get() });
```

Comment:
maybe token can be automatic from ArticleApi construuctor if not specified? see api-base class suggestion.

Reply:
its calling super(options) in constructor should be fine?